tests/test_isdtype.py

```diff
@@ -61,12 +59,12 @@ def isdtype_(dtype_, kind):
         res = dtype_categories[kind](dtype_)
     else:
         res = dtype_ == kind
-    assert type(res) is bool
+    assert isinstance(res, bool)
     return res
 
 @pytest.mark.parametrize("library", ["cupy", "numpy", "torch"])
 def test_isdtype_spec_dtypes(library):
-    xp = import_('array_api_compat.' + library)
+    xp = pytest.importorskip('array_api_compat.' + library)
```

Comment:
Why did you change this? This is not equivalent to the import_ helper.

Reply:
And the same elsewhere. The only tests that should be skipped are cupy tests, because cupy cannot be installed everywhere (and in particular, not on CI).

Reply:
OK, I see. I looked at the dependency definitions and could not find torch as required or extra. So I assumed it was optional and test can be skipped.

Reply:
By default those tests failed for me, because torch was not installed.

Reply:
All runtime dependencies are optional. But for the tests, we want to require all optional dependencies so that we make sure the tests are actually run. The exception is cupy, which simply cannot be installed everywhere. 

I'm not sure if we can express this in a better way in the package metadata.

Reply:
One could add another entry in the `extras_require` dict, like `{"test-no-cupy": ["numpy", "torch"], "test": ["numpy", "torch", "cupy"]}`. But I would suggest to not do this in this PR. If the package grows further, it is probably worth using `tox` or similar to handle the different environments.